.changeset/grumpy-lies-bathe.md

```diff
@@ -0,0 +1,11 @@
+---
+"@salt-ds/core": minor
+---
+
+Added SemanticIconProvider to provide a mechanism to swap out built-in component icons to custom icons.
```

Comment:
Apart from changelog, we will need a new doc page on the website, talking about how to use the provider, as well as how / requirements of custom icon sets.

Reply:
added